---
title: "RNAseq_final_KS1-KS2"
author: "Christine Gao"
date: "7/5/2023"
output: 
  html_document:
    code_folding: hide
---

```{r, include=FALSE}
library(TxDb.Mmusculus.UCSC.mm10.ensGene)
library(tximport)
library(DESeq2)
library(sva)
library(biomaRt)
library(ggplot2)
library(ggrepel)
library(tximeta)
library(SummarizedExperiment)
library(edgeR)
library(limma)
library(statmod)
library(tidyverse)
library(scales)
```

Function for annotating diffexp results
```{r}
ensembl.annotate <- function(diffexp.subset){
  ensembl.id <- rownames(diffexp.subset)

  ensembl102.mmusculus <- useEnsembl(biomart='genes', dataset='mmusculus_gene_ensembl', version=102)
  mgi <- getBM(attributes=c('ensembl_gene_id','mgi_symbol','mgi_description'),
               filters='ensembl_gene_id',
               values=ensembl.id,
               mart=ensembl102.mmusculus)
  diffexp.subset$ensembl_gene_id <- ensembl.id
  diffexp.subset <- merge(diffexp.subset, mgi, by='ensembl_gene_id')
  diffexp.subset <- diffexp.subset[order(diffexp.subset$log2FoldChange, decreasing=TRUE),]
  return(diffexp.subset)
}
```

Function for retrieving gene names, given a vector of Ensembl IDs. Returns a dataframe containing both Ensembl IDs and corresponding genes. <i>**Leaves out Ensembl IDs for which a gene name could not be found.</i>
```{r}
get.gene <- function(x){
  ensembl102.mmusculus <- useEnsembl(biomart='genes', dataset='mmusculus_gene_ensembl', version=102)
  mgi <- getBM(attributes=c('ensembl_gene_id','mgi_symbol'),
               filters='ensembl_gene_id',
               values=x,
               mart=ensembl102.mmusculus)
  genes <- data.frame(ensembl_gene_id=x)
  genes <- merge(genes, mgi, by='ensembl_gene_id')
  return(genes)
}
```

Sample sheet: all samples
```{r}
samples <- read.csv('RNAseq-samples_230510.csv')
```

Sample sheet: KS2, D14 only
```{r}
samples <- read.csv('RNAseq-samples_230510_KS2-D14-only.csv')
```

Sample sheet: KS2, D7 only
```{r}
samples <- read.csv('RNAseq-samples_230510_KS2-D7-only.csv')
```

Sample sheet: KS1, D14 only
```{r}
samples <- read.csv('RNAseq-samples_230510_KS1-D14-only.csv')
```

Sample sheet: KS1, D7 only
```{r}
samples <- read.csv('RNAseq-samples_230510_KS1-D7-only.csv')
```

Sample sheet: KS2, D14, combined Wt
```{r}
samples <- read.csv('RNAseq-samples_230510_KS2-D14-comboWt.csv')
```

Sample sheet: KS1, D14, combined Wt
```{r}
samples <- read.csv('RNAseq-samples_230510_KS1-D14-comboWt.csv')
```

Sample sheet: KS1 mut, D7 and D14
```{r}
samples <- read.csv('RNAseq-samples_230510_KS1_days.csv')
```

Sample sheet: KS1 mut, D7 and D14
```{r}
samples <- read.csv('RNAseq-samples_230510_KS1_days.csv')
```

Sample sheet: KS2 mut, D7 and D14
```{r}
samples <- read.csv('RNAseq-samples_230510_KS2_days.csv')
```

Sample sheet: KS1 Wt, D7 and D14
```{r}
samples <- read.csv('RNAseq-samples_230510_KS1_Wt-days.csv')
```

Sample sheet: KS2 Wt, D7 and D14
```{r}
samples <- read.csv('RNAseq-samples_230510_KS2_Wt-days.csv')
```

Import quant files and summarize to gene level
```{r}
dir <- '~/Desktop/temp/Fahrner lab/Data/KS2/230510_KS1-KS2_NGS/quants_final_KS1-KS2'
files <- file.path(dir, paste(samples$Sample.ID, '_quant', sep=''), 'quant.sf')
file.exists(files)
coldata <- data.frame(files, 
                      names=samples$Sample.ID, 
                      genotype=as.factor(samples$Genotype), 
                      line=as.factor(samples$Line),
                      kabuki=as.factor(samples$Kabuki),
                      day=as.factor(samples$Day),
                      stringsAsFactors = FALSE)

# import transcript abundances
se <- tximeta(coldata)

# summarize abundances to gene level
gse <- summarizeToGene(se)
```

Set design formula and create DESeqDataSet: all samples
```{r}
dds <- DESeqDataSet(gse, design=~genotype*kabuki*day)
```

Set design formula and create DESeqDataSet: by genotype
```{r}
dds <- DESeqDataSet(gse, design=~genotype)
```

Set design formula and create DESeqDataSet: by day
```{r}
dds <- DESeqDataSet(gse, design=~day)
```

Collapse replicates and filter genes for low counts
```{r}
# collapse technical replicates
dds$sample <- factor(gsub("_S.*", "", colnames(dds)))
dds.coll <- collapseReplicates(dds, dds$sample)
stopifnot(all(rowSums(counts(dds[,which(dds$sample==dds$sample[1])])) ==
                counts(dds.coll[,which(colnames(dds.coll)==dds$sample[1])])))
dds.coll.counts <- counts(dds.coll)

# filter genes with low median counts
keep <- rowMedians(counts(dds.coll)) > 10
dds.coll.filtered <- dds.coll[keep,]
```

Fig 5A: PCA plot: all samples
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('genotype','kabuki','day'), returnData=TRUE)

plot$group <- factor(plot$group, levels=c('Wt:KS1:D7','Wt:KS1:D14',
                                          'Mut:KS1:D7','Mut:KS1:D14',
                                          'Wt:KS2:D7','Wt:KS2:D14',
                                          'Mut:KS2:D7','Mut:KS2:D14'))
group.colors <- c('#0F80FF','#0F80FF',
                  '#FB0207','#FB0207',
                  '#0F80FF','#0F80FF',
                  '#FB0207','#FB0207')
group.shapes <- c(1,2,
                  1,2,
                  16,17,
                  16,17)
# setEPS()
# postscript('plots_final/pca_all.eps', width=2.15, height=1.57)
ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 2) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 32% variance') +
  ylab('PC2: 18% variance') +
  theme_bw(base_size=8) +
  theme(axis.text=element_text(size=8, color='black'),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_rect(color='black',
                                    fill=NA,
                                    size=1),
        legend.position = 'none')
# dev.off()
```

PCA plot: KS2, D14 only
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('genotype'), returnData=TRUE)
plot$group <- factor(plot$group, levels=c('Wt', 'Mut'))
group.colors <- c('#0F80FF','#FB0207')
group.shapes <- c(16,17)

ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 3) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 40% variance') +
  ylab('PC2: 28% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

PCA plot: KS1, D14 only
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('genotype'), returnData=TRUE)
plot$group <- factor(plot$group, levels=c('Wt', 'Mut'))
group.colors <- c('#0F80FF','#FB0207')
group.shapes <- c(16,17)

ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 3) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 56% variance') +
  ylab('PC2: 24% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

PCA plot: KS1, D7 only
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('genotype'), returnData=TRUE)
plot$group <- factor(plot$group, levels=c('Wt', 'Mut'))
group.colors <- c('#0F80FF','#FB0207')
group.shapes <- c(16,17)

ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 3) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 59% variance') +
  ylab('PC2: 27% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

PCA plot: KS2, D14, combined Wt
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('genotype'), returnData=TRUE)
plot$group <- factor(plot$group, levels=c('Wt', 'Mut'))
group.colors <- c('#0F80FF','#FB0207')
group.shapes <- c(16,17)

ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 3) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 32% variance') +
  ylab('PC2: 27% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

PCA plot: KS1, D14, combined Wt
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('genotype'), returnData=TRUE)
plot$group <- factor(plot$group, levels=c('Wt', 'Mut'))
group.colors <- c('#0F80FF','#FB0207')
group.shapes <- c(16,17)

ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 3) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 37% variance') +
  ylab('PC2: 27% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

PCA plot: KS1 D14 and KS1 D7
```{r}
plot <- plotPCA(vst(dds.coll.filtered), intgroup=c('day'), returnData=TRUE)
plot$group <- factor(plot$group, levels=c('D7', 'D14'))
group.colors <- c('#FB0207','#FB0207')
group.shapes <- c(16,17)

ggplot(data = plot, aes(x = PC1, y = PC2, color=group, shape=group)) + 
  geom_point(size = 3) + 
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) + 
  xlab('PC1: 59% variance') +
  ylab('PC2: 35% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())
```

Analysis: KS2, D14 only
```{r}
dds.coll.filtered$genotype <- relevel(dds.coll.filtered$genotype, ref="Wt")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~genotype, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
design(ddssva) <- formula(~SV1 + genotype)

ddssva <- DESeq(ddssva)
dds <- ddssva

# save(dds, file='dds_230510_KS2-D14-only.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
# write.csv(diffexp.subset, file='diffexp_230510_KS2-D14-only.csv')
```

Analysis: KS2, D7 only
```{r}
dds.coll.filtered$genotype <- relevel(dds.coll.filtered$genotype, ref="Wt")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~genotype, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
design(ddssva) <- formula(~SV1 + genotype)

ddssva <- DESeq(ddssva)
dds <- ddssva

save(dds, file='dds_230510_KS2-D7-only.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
write.csv(diffexp.subset, file='diffexp_230510_KS2-D7-only.csv')
```

Analysis: KS1, D14 only
```{r}
dds.coll.filtered$genotype <- relevel(dds.coll.filtered$genotype, ref="Wt")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~genotype, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
ddssva$SV2 <- svseq$sv[,2]
design(ddssva) <- formula(~SV1 + SV2 + genotype)

ddssva <- DESeq(ddssva)
dds <- ddssva

# save(dds, file='dds_230510_KS1-D14-only.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
# write.csv(diffexp.subset, file='diffexp_230510_KS1-D14-only.csv')
```

Analysis: KS1, D7 only
```{r}
dds.coll.filtered$genotype <- relevel(dds.coll.filtered$genotype, ref="Wt")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~genotype, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
design(ddssva) <- formula(~SV1 + genotype)

ddssva <- DESeq(ddssva)
dds <- ddssva

# save(dds, file='dds_230510_KS1-D7-only.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
# write.csv(diffexp.subset, file='diffexp_230510_KS1-D7-only.csv')
```

Analysis: KS2, D14, combined Wt
```{r}
dds.coll.filtered$genotype <- relevel(dds.coll.filtered$genotype, ref="Wt")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~genotype, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
ddssva$SV2 <- svseq$sv[,2]
design(ddssva) <- formula(~SV1 + SV2 + genotype)

ddssva <- DESeq(ddssva)
dds <- ddssva

# save(dds, file='dds_230510_KS2-D14-comboWt.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
# write.csv(diffexp.subset, file='diffexp_230510_KS2-D14-comboWt.csv')
```

Analysis: KS1, D14, combined Wt
```{r}
dds.coll.filtered$genotype <- relevel(dds.coll.filtered$genotype, ref="Wt")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~genotype, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
ddssva$SV2 <- svseq$sv[,2]
design(ddssva) <- formula(~SV1 + SV2 + genotype)

ddssva <- DESeq(ddssva)
dds <- ddssva

# save(dds, file='dds_230510_KS1-D14-comboWt.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
# write.csv(diffexp.subset, file='diffexp_230510_KS1-D14-comboWt.csv')
```

Analysis: KS1 D14 vs D7
```{r}
dds.coll.filtered$day <- relevel(dds.coll.filtered$day, ref="D7")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~day, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
design(ddssva) <- formula(~SV1 + day)

ddssva <- DESeq(ddssva)
dds <- ddssva

save(dds, file='dds_230510_KS1_day.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
write.csv(diffexp.subset, file='diffexp_230510_KS1_day.csv')
```

Analysis: KS1 Wt D14 vs D7
```{r}
dds.coll.filtered$day <- relevel(dds.coll.filtered$day, ref="D7")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~day, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
design(ddssva) <- formula(~SV1 + day)

ddssva <- DESeq(ddssva)
dds <- ddssva

save(dds, file='dds_230510_KS1_Wt-day.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
write.csv(diffexp.subset, file='diffexp_230510_KS1_Wt-day.csv')
```

Analysis: KS2 Wt D14 vs D7
```{r}
dds.coll.filtered$day <- relevel(dds.coll.filtered$day, ref="D7")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~day, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
ddssva$SV2 <- svseq$sv[,2]
design(ddssva) <- formula(~SV1 + SV2 + day)

ddssva <- DESeq(ddssva)
dds <- ddssva

save(dds, file='dds_230510_KS2_Wt-day.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
write.csv(diffexp.subset, file='diffexp_230510_KS2_Wt-day.csv')
```

Analysis: KS2 D14 vs D7
```{r}
dds.coll.filtered$day <- relevel(dds.coll.filtered$day, ref="D7")
dds.coll.filtered.counts <- counts(dds.coll.filtered)
mod <- model.matrix(~day, colData(dds.coll.filtered))
mod0 <- model.matrix(~1, colData(dds.coll.filtered))
svseq <- svaseq(dds.coll.filtered.counts, mod, mod0)

ddssva <- dds.coll.filtered
ddssva$SV1 <- svseq$sv[,1]
design(ddssva) <- formula(~SV1 + day)

ddssva <- DESeq(ddssva)
dds <- ddssva

save(dds, file='dds_230510_KS2_day.rda')
res <- results(dds)
diffexp.subset <- as.data.frame(res[which(res$padj <0.1),])

diffexp.subset <- ensembl.annotate(diffexp.subset)
write.csv(diffexp.subset, file='diffexp_230510_KS2_day.csv')
```

Supp Fig 9C: Conditional p-val hist of KS2 D14 p-val, stratified by significance in prior KS2 exp
```{r}
load(file='dds_230510_KS2-D14-only.rda')
res.230510.KS2.D14 <- results(dds)

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/BigStuff/dds_final_no-sva.rda')
res.old <- results(dds)

sig.res.old <- rownames(unique(res.old[which(res.old$padj<0.1),]))

# setEPS()
# postscript(file='plots_final/density-cond_KS2-old.eps', width=1.9, height=1.57)
par(mar=c(1.35,1.1,0,0), mgp=c(0.5,0.075,0))
plot(density(res.230510.KS2.D14[which(rownames(res.230510.KS2.D14) %in% sig.res.old),]$pvalue, 
             from=0.001,to=0.99, bw=0.025), 
     asp=NA, lwd=2, col='orange', main='', bty='n', yaxt='n', xaxt='n', ylab='', xlab='')

axis(1, at=c(0,1), tck=-0.02, cex.axis=0.7)
axis(2, at=c(0,10), tck=-0.02, cex.axis=0.7, pos=-0.02)
mtext('p-values, KS2 D14', side=1, line=0.3, cex=0.7)
mtext('Density', side=2, line=0.3, cex=0.7)

lines(density(res.230510.KS2.D14[which(!(rownames(res.230510.KS2.D14) %in% sig.res.old)),]$pvalue, 
              from=0.001,to=0.99, bw=0.025),
      lwd=2, col='black')

legend('topright', legend=c('p-adj > 0.1, prior KS2', 'p-adj < 0.1, prior KS2'),
       col=c("black", "orange"), lty=c(1,1), lwd=2, cex=0.7,
       box.lty=0)
# dev.off()
```

Supp Fig 9D: Scatter plot comparing log2FC from KS2, D14 only with prior KS2 experiment
```{r}
shared <- unique(c(rownames(res.230510.KS2.D14[which(res.230510.KS2.D14$padj<0.1),]),
                   rownames(res.old[which(res.old$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS2.D14)) &
                         (shared %in% rownames(res.old)))]
shared.df <- data.frame(ensembl=shared,
                        log2FC_230510=res.230510.KS2.D14[shared,]$log2FoldChange,
                        padj_230510=res.230510.KS2.D14[shared,]$padj,
                        log2FC_old=res.old[shared,]$log2FoldChange,
                        padj_old=res.old[shared,]$padj)
shared.sig.df <- shared.df[which(shared.df$padj_230510<0.1 &
                                   shared.df$padj_old<0.1),]
# setEPS()
# postscript(file='plots_final/log2FC_KS2-old.eps', width=2.26, height=1.57)
cairo_pdf(file='plots_final/log2FC_KS2-old.pdf', width=2.26, height=1.57,
          onefile = FALSE, fallback_resolution = 600)
par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0.25,0))
plot(shared.df$log2FC_230510, shared.df$log2FC_old,
     pch=16, cex=0.4, col=alpha('gray60',0.3),
     xlim=c(-10,10), ylim=c(-8,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')
points(shared.sig.df$log2FC_230510, shared.sig.df$log2FC_old, 
       pch=16, cex=0.4, col=alpha('orange',0.7))
axis(1, seq(-10,10, by=2), tck=-0.02, cex.axis=0.7)
axis(2, seq(-8,8, by=2), tck=-0.015, cex.axis=0.7)
mtext('log2FC, KS2 D14', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, prior KS2', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
dev.off()
```

Fig 5B: Conditional p-val hist of KS1 D14 p-val, stratified by KS2 D14 significance
```{r}
load(file='dds_230510_KS1-D14-only.rda')
res.230510.KS1.D14 <- results(dds)

load(file='dds_230510_KS2-D14-only.rda')
res.230510.KS2.D14 <- results(dds)

sig.res.KS2 <- rownames(unique(res.230510.KS2.D14[which(res.230510.KS2.D14$padj<0.1),]))

# setEPS()
# postscript(file='plots_final/density-cond_KS1-KS2-D14.eps', width=1.9, height=1.57)
par(mar=c(1.35,1.1,0,0), mgp=c(0.5,0.075,0))
plot(density(res.230510.KS1.D14[which(rownames(res.230510.KS1.D14) %in% sig.res.KS2),]$pvalue, 
             from=0.001,to=0.99, bw=0.025), 
     asp=NA, lwd=2, col='orange', main='', bty='n', yaxt='n', xaxt='n', ylab='', xlab='')

axis(1, at=c(0,1), tck=-0.02, cex.axis=0.7)
axis(2, at=c(0.2,6), tck=-0.02, cex.axis=0.7, pos=-0.02)
mtext('p-values, KS1 D14', side=1, line=0.3, cex=0.7)
mtext('Density', side=2, line=0.3, cex=0.7)

lines(density(res.230510.KS1.D14[which(!(rownames(res.230510.KS1.D14) %in% sig.res.KS2)),]$pvalue, 
              from=0.001,to=0.99, bw=0.025),
      lwd=2, col='black')

legend('topright', legend=c('p-adj > 0.1, KS2 D14', 'p-adj < 0.1, KS2 D14'),
       col=c("black", "orange"), lty=c(1,1), lwd=2, cex=0.7,
       box.lty=0)
# dev.off()
```

Fig 5C: Scatter plot comparing log2FC from KS2, D14 with KS1, D14
```{r}
shared <- unique(c(rownames(res.230510.KS2.D14[which(res.230510.KS2.D14$padj<0.1),]),
                   rownames(res.230510.KS1.D14[which(res.230510.KS1.D14$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS2.D14)) &
                         (shared %in% rownames(res.230510.KS1.D14)))]

shared.df <- data.frame(ensembl_gene_id=shared,
                        log2FC_KS2=res.230510.KS2.D14[shared,]$log2FoldChange,
                        padj_KS2=res.230510.KS2.D14[shared,]$padj,
                        log2FC_KS1=res.230510.KS1.D14[shared,]$log2FoldChange,
                        padj_KS1=res.230510.KS1.D14[shared,]$padj)

genes <- get.gene(shared.df$ensembl_gene_id)
shared.df <- merge(shared.df, genes, by='ensembl_gene_id')

# shared.sig.df <- shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]
# shared.sig.df$sign_KS2 <- sign(shared.sig.df$log2FC_KS2)
# shared.sig.df$sign_KS1 <- sign(shared.sig.df$log2FC_KS1)
# shared.sig.df$sum <- rowSums(shared.sig.df[,c('sign_KS2','sign_KS1')])
# genes <- get.gene(shared.sig.df$ensembl)
# shared.sig.df <- merge(shared.sig.df, genes, by='ensembl_gene_id')
# write.csv(shared.sig.df, file='shared.sig.df.csv')

# setEPS()
# postscript(file='plots_final/log2FC_KS1-KS2-D14.eps', width=2.26, height=1.57)
# cairo_pdf(filename='plots_final/log2FC_KS1-KS2-D14.pdf', width=2.26, height=1.57,
#          onefile=FALSE, fallback_resolution = 600)
par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0.25,0))
plot(shared.df$log2FC_KS2, shared.df$log2FC_KS1,
     pch=16, cex=0.4, col=alpha('gray60',0.3),
     xlim=c(-10,10), ylim=c(-8,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/MGI.chon.rda')

points(shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS2,
       shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS1,
       pch=16, cex=0.4, col=alpha('orange',0.7))

points.chon <- shared.df[which(shared.df$padj_KS2<0.1 & 
                                 shared.df$padj_KS1<0.1 & 
                                 shared.df$ensembl %in% MGI.chon$gene_id),]

points(points.chon$log2FC_KS2, points.chon$log2FC_KS1,
       pch=16, cex=0.4, col='deepskyblue')

# text(points.chon$log2FC_KS2+0.2, 
#      points.chon$log2FC_KS1, 
#      adj=0, cex = 0.6, col='black',font=3,
#      labels=points.chon$mgi_symbol)

int.genes <- c('Acan','Col11a1','Col10a1','Matn1','Matn3')

points(shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS2,
       shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS1,
       pch = 1, cex = 1, col = 'black')

axis(1, seq(-10,10, by=2), tck=-0.02, cex.axis=0.7)
axis(2, seq(-8,8, by=2), tck=-0.015, cex.axis=0.7)
mtext('log2FC, KS2 D14', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, KS1 D14', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')

xJitter <- c(-3.8,-5.5,0.7,0.7,-4.5)
yJitter <- c(1.8,0.5,0,0,-0.6)
text(shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS2 + xJitter,
     shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS1 + yJitter,
     adj=0, cex = 0.6, col='black',font=3,
     labels=shared.df[which(shared.df$mgi_symbol %in% int.genes),]$mgi_symbol)
# dev.off()
```

Supp Fig 9A: Conditional p-val hist of KS1 D7 p-val, stratified by prior KS1 experiment
```{r}
load(file='dds_230510_KS1-D7-only.rda')
res.230510.KS1.D7 <- results(dds)

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/BigStuff/dds_KS1.rda')
res.old.KS1 <- results(dds)

sig.res.old.KS1 <- rownames(unique(res.old.KS1[which(res.old.KS1$padj<0.1),]))

# setEPS()
# postscript(file='plots_final/density-cond_KS1-old.eps', width=1.9, height=1.57)
par(mar=c(1.35,1.1,0,0), mgp=c(0.5,0.075,0))
plot(density(res.230510.KS1.D7[which(rownames(res.230510.KS1.D7) %in% sig.res.old.KS1),]$pvalue, 
             from=0.001,to=0.99, bw=0.025), 
     asp=NA, lwd=2, col='orange', main='', bty='n', yaxt='n', xaxt='n', ylab='', xlab='')

axis(1, at=c(0,1), tck=-0.02, cex.axis=0.7)
axis(2, at=c(0,11), tck=-0.02, cex.axis=0.7, pos=-0.02)
mtext('p-values, KS1 D7', side=1, line=0.3, cex=0.7)
mtext('Density', side=2, line=0.3, cex=0.7)

lines(density(res.230510.KS1.D7[which(!(rownames(res.230510.KS1.D7) %in% sig.res.old.KS1)),]$pvalue, 
              from=0.001,to=0.99, bw=0.025),
      lwd=2, col='black')

legend('topright', legend=c('p-adj > 0.1, prior KS1', 'p-adj < 0.1, prior KS1'),
       col=c("black", "orange"), lty=c(1,1), lwd=2, cex=0.7,
       box.lty=0)
# dev.off()
```

Supp Fig 9B: Scatter plot comparing log2FC from KS1, D7 with prior KS1 experiment
```{r}
shared <- unique(c(rownames(res.230510.KS1.D7[which(res.230510.KS1.D7$padj<0.1),]),
                   rownames(res.old.KS1[which(res.old.KS1$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS1.D7)) &
                         (shared %in% rownames(res.old.KS1)))]
shared.df <- data.frame(ensembl=shared,
                        log2FC_230510=res.230510.KS1.D7[shared,]$log2FoldChange,
                        padj_230510=res.230510.KS1.D7[shared,]$padj,
                        log2FC_old=res.old.KS1[shared,]$log2FoldChange,
                        padj_old=res.old.KS1[shared,]$padj)
shared.sig.df <- shared.df[which(shared.df$padj_230510<0.1 &
                                   shared.df$padj_old<0.1),]

# setEPS()
# postscript(file='plots_final/log2FC_KS1-old.eps', width=2.26, height=1.57)
cairo_pdf(file='plots_final/log2FC_KS1-old.pdf', width=2.26, height=1.57,
          onefile = FALSE, fallback_resolution = 600)
par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0.25,0))
plot(shared.df$log2FC_230510, shared.df$log2FC_old,
     pch=16, cex=0.4, col=alpha('gray60',0.3),
     xlim=c(-10,10), ylim=c(-10,10),
     xlab = '', ylab='', axes=FALSE, 
     main='')
points(shared.sig.df$log2FC_230510, shared.sig.df$log2FC_old, 
       pch=16, cex=0.4, col=alpha('orange',0.7))
axis(1, seq(-10,10, by=2), tck=-0.02, cex.axis=0.7)
axis(2, seq(-10,10, by=2), tck=-0.015, cex.axis=0.7)
mtext('log2FC, KS1 D7', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, prior KS1', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
dev.off()
```

Conditional p-val hist of KS2 D14 comboWt p-val, stratified by prior KS2 experiment
```{r}
load('dds_230510_KS2-D14-comboWt.rda')
res.230510.KS2.D14.combo <- results(dds)

load('~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/BigStuff/dds_final_no-sva.rda')
res.old <- results(dds)

sig.res.old <- rownames(unique(res.old[which(res.old$padj<0.1),]))

plot(density(res.230510.KS2.D14.combo[which(rownames(res.230510.KS2.D14.combo) 
                                            %in% sig.res.old),]$pvalue, 
             from=0.001,to=0.99, bw=0.025), 
     asp=NA, lwd=2, col='orange', main='', bty='n', yaxt='n', xaxt='n', ylab='', xlab='')

axis(1, at=c(0,1), tck=-0.02, cex.axis=0.7)
axis(2, at=c(0,8), tck=-0.02, cex.axis=0.7, pos=-0.02)
mtext('p-values, 230510 KS2 D14 comboWt', side=1, line=0.3, cex=0.7)
mtext('Density', side=2, line=0.3, cex=0.7)

lines(density(res.230510.KS2.D14.combo[which(!(rownames(res.230510.KS2.D14.combo) 
                                               %in% sig.res.old)),]$pvalue, 
              from=0.001,to=0.99, bw=0.025),
      lwd=2, col='black')

legend('topright', legend=c('p-adj > 0.1, old exp', 'p-adj < 0.1, old exp'),
       col=c("black", "orange"), lty=c(1,1), lwd=2, cex=0.7,
       box.lty=0)
```

Scatter plot comparing log2FC from KS2 D14 comboWt with prior KS2 experiment
```{r}
shared <- unique(c(rownames(res.230510.KS2.D14.combo[which(res.230510.KS2.D14.combo$padj<0.1),]),
                   rownames(res.old[which(res.old$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS2.D14.combo)) &
                         (shared %in% rownames(res.old)))]
shared.df <- data.frame(ensembl=shared,
                        log2FC_230510=res.230510.KS2.D14.combo[shared,]$log2FoldChange,
                        padj_230510=res.230510.KS2.D14.combo[shared,]$padj,
                        log2FC_old=res.old[shared,]$log2FoldChange,
                        padj_old=res.old[shared,]$padj)
shared.sig.df <- shared.df[which(shared.df$padj_230510<0.1 &
                                   shared.df$padj_old<0.1),]

par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0.25,0))
plot(shared.df$log2FC_230510, shared.df$log2FC_old,
     pch=16, cex=0.5, col='gray60',
     xlim=c(-10,10), ylim=c(-8,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')
points(shared.sig.df$log2FC_230510, shared.sig.df$log2FC_old, 
       pch=16, cex=0.5, col='orange')
axis(1, seq(-10,10, by=2), tck=-0.02, cex.axis=0.7)
axis(2, seq(-8,8, by=2), tck=-0.015, cex.axis=0.7)
mtext('log2FC, 230510 KS2 D14 comboWt', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, old exp', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
```

Conditional p-val hist of KS1 D14 comboWt, stratified by KS2 D14 comboWt sig
```{r}
load('dds_230510_KS2-D14-comboWt.rda')
res.230510.KS2.D14.combo <- results(dds)

load('dds_230510_KS1-D14-comboWt.rda')
res.230510.KS1.D14.combo <- results(dds)

sig.res.KS2.combo <- 
  rownames(unique(res.230510.KS2.D14.combo[which(res.230510.KS2.D14.combo$padj<0.1),]))

plot(density(res.230510.KS1.D14.combo[which(rownames(res.230510.KS1.D14.combo) 
                                            %in% sig.res.KS2.combo),]$pvalue, 
             from=0.001,to=0.99, bw=0.025), 
     asp=NA, lwd=2, col='orange', main='', bty='n', yaxt='n', xaxt='n', ylab='', xlab='')

axis(1, at=c(0,1), tck=-0.02, cex.axis=0.7)
axis(2, at=c(0.2,5.5), tck=-0.02, cex.axis=0.7, pos=-0.02)
mtext('p-values, 230510 KS1 D14 comboWt', side=1, line=0.3, cex=0.7)
mtext('Density', side=2, line=0.3, cex=0.7)

lines(density(res.230510.KS1.D14.combo[which(!(rownames(res.230510.KS1.D14.combo) 
                                               %in% sig.res.KS2.combo)),]$pvalue, 
              from=0.001,to=0.99, bw=0.025),
      lwd=2, col='black')

legend('topright', legend=c('p-adj > 0.1, KS2 D14 comboWt', 'p-adj < 0.1, KS2 D14 comboWt'),
       col=c("black", "orange"), lty=c(1,1), lwd=2, cex=0.7,
       box.lty=0)
```

Scatter plot comparing log2FC from KS2 D14 comboWt with KS1 D14 comboWt
```{r}
shared <- unique(c(rownames(res.230510.KS2.D14.combo[which(res.230510.KS2.D14.combo$padj<0.1),]),
                   rownames(res.230510.KS1.D14.combo[which(res.230510.KS1.D14.combo$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS2.D14.combo)) &
                         (shared %in% rownames(res.230510.KS1.D14.combo)))]

shared.df <- data.frame(ensembl=shared,
                        log2FC_KS2=res.230510.KS2.D14.combo[shared,]$log2FoldChange,
                        padj_KS2=res.230510.KS2.D14.combo[shared,]$padj,
                        log2FC_KS1=res.230510.KS1.D14.combo[shared,]$log2FoldChange,
                        padj_KS1=res.230510.KS1.D14.combo[shared,]$padj)

par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0.25,0))
plot(shared.df$log2FC_KS2, shared.df$log2FC_KS1,
     pch=16, cex=0.5, col='gray60',
     xlim=c(-10,10), ylim=c(-8,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/MGI.chon.rda')

# points(shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS2,
#        shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS1,
#        pch=16, cex=0.5, col='orange')

points(shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1 & shared.df$ensembl %in% MGI.chon$gene_id),]$log2FC_KS2,
       shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1 & shared.df$ensembl %in% MGI.chon$gene_id),]$log2FC_KS1,
       pch=16, cex=0.5, col='deepskyblue')

axis(1, seq(-10,10, by=2), tck=-0.02, cex.axis=0.7)
axis(2, seq(-8,8, by=2), tck=-0.015, cex.axis=0.7)
mtext('log2FC, 230510 KS2 D14 comboWt', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, 230510 KS1 D14 comboWt', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
```

Fig 5D: Conditional p-val hist of KS1 D7, stratified by KS2 D7 sig
```{r}
load('dds_230510_KS2-D7-only.rda')
res.230510.KS2.D7 <- results(dds)

load('dds_230510_KS1-D7-only.rda')
res.230510.KS1.D7 <- results(dds)

sig.res.KS2 <- 
  rownames(unique(res.230510.KS2.D7[which(res.230510.KS2.D7$padj<0.1),]))

# setEPS()
# postscript(file='plots_final/density-cond_KS1-KS2-D7.eps', width=1.9, height=1.57)
par(mar=c(1.35,1.1,0.3,0), mgp=c(0.5,0.075,0))
plot(density(res.230510.KS1.D7[which(rownames(res.230510.KS1.D7)
                                     %in% sig.res.KS2),]$pvalue, 
             from=0.001,to=0.99, bw=0.025), 
     asp=NA, lwd=2, col='orange', main='', bty='n', yaxt='n', xaxt='n', ylab='', xlab='')

axis(1, at=c(0,1), tck=-0.02, cex.axis=0.7)
axis(2, at=c(0.2,6), tck=-0.02, cex.axis=0.7, pos=-0.02)
mtext('p-values, KS1 D7', side=1, line=0.3, cex=0.7)
mtext('Density', side=2, line=0.3, cex=0.7)

lines(density(res.230510.KS1.D7[which(!(rownames(res.230510.KS1.D7) 
                                               %in% sig.res.KS2)),]$pvalue, 
              from=0.001,to=0.99, bw=0.025),
      lwd=2, col='black')

legend('topright', legend=c('p-adj > 0.1, KS2 D7', 'p-adj < 0.1, KS2 D7'),
       col=c("black", "orange"), lty=c(1,1), lwd=2, cex=0.7,
       box.lty=0)
# dev.off()
```

Fig 5E: Scatter plot comparing log2FC from KS1 D7 with KS2 D7
```{r}
shared <- unique(c(rownames(res.230510.KS2.D7[which(res.230510.KS2.D7$padj<0.1),]),
                   rownames(res.230510.KS1.D7[which(res.230510.KS1.D7$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS2.D7)) &
                         (shared %in% rownames(res.230510.KS1.D7)))]

shared.df <- data.frame(ensembl_gene_id=shared,
                        log2FC_KS2=res.230510.KS2.D7[shared,]$log2FoldChange,
                        padj_KS2=res.230510.KS2.D7[shared,]$padj,
                        log2FC_KS1=res.230510.KS1.D7[shared,]$log2FoldChange,
                        padj_KS1=res.230510.KS1.D7[shared,]$padj)

genes <- get.gene(shared.df$ensembl_gene_id)
shared.df <- merge(shared.df, genes, by='ensembl_gene_id')

# setEPS()
# postscript(file='plots_final/log2FC_KS1-KS2-D7.eps', width=2.26, height=1.57)
cairo_pdf(filename='plots_final/log2FC_KS1-KS2-D7.pdf', width=2.26, height=1.57,
         onefile=FALSE, fallback_resolution = 600)
par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0.25,0))
plot(shared.df$log2FC_KS2, shared.df$log2FC_KS1,
     pch=16, cex=0.4, col=alpha('gray60',0.3),
     xlim=c(-10,10), ylim=c(-8,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/MGI.chon.rda')

points(shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS2,
       shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS1,
       pch=16, cex=0.4, col=alpha('orange',0.7))

points.chon <- shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1 & shared.df$ensembl %in% MGI.chon$gene_id),]

points(points.chon$log2FC_KS2, points.chon$log2FC_KS1,
       pch=16, cex=0.4, col='deepskyblue')

int.genes <- c('Acan','Col2a1','Col11a1','Col10a1')

points(points.chon[which(points.chon$mgi_symbol %in% int.genes),]$log2FC_KS2,
       points.chon[which(points.chon$mgi_symbol %in% int.genes),]$log2FC_KS1,
       pch = 1, cex = 1, col = 'black')

xJitter <- c(-4.6,1,0.7,1)
yJitter <- c(0,0,0.5,0)

text(points.chon[which(points.chon$mgi_symbol %in% int.genes),]$log2FC_KS2+xJitter, 
     points.chon[which(points.chon$mgi_symbol %in% int.genes),]$log2FC_KS1+yJitter, 
     adj=0, cex = 0.6, col='black',font=3,
     labels=points.chon[which(points.chon$mgi_symbol %in% int.genes),]$mgi_symbol)

axis(1, seq(-10,10, by=2), tck=-0.02, cex.axis=0.7)
axis(2, seq(-8,8, by=2), tck=-0.015, cex.axis=0.7)
mtext('log2FC, KS2 D7', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, KS1 D7', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
dev.off()
```

Fig 5F: Scatter plot comparing log2FC from KS1 D14 vs 7 with KS2 D14 vs 7
```{r}
load('dds_230510_KS2_day.rda')
res.230510.KS2.days <- results(dds)

load('dds_230510_KS1_day.rda')
res.230510.KS1.days <- results(dds)

shared <- unique(c(rownames(res.230510.KS2.days[which(res.230510.KS2.days$padj<0.1),]),
                   rownames(res.230510.KS1.days[which(res.230510.KS1.days$padj<0.1),])))
shared <- shared[which((shared %in% rownames(res.230510.KS2.days)) &
                         (shared %in% rownames(res.230510.KS1.days)))]

shared.df <- data.frame(ensembl_gene_id=shared,
                        log2FC_KS2=res.230510.KS2.days[shared,]$log2FoldChange,
                        padj_KS2=res.230510.KS2.days[shared,]$padj,
                        log2FC_KS1=res.230510.KS1.days[shared,]$log2FoldChange,
                        padj_KS1=res.230510.KS1.days[shared,]$padj)

genes <- get.gene(shared.df$ensembl_gene_id)
shared.df <- merge(shared.df, genes, by='ensembl_gene_id')

# setEPS()
# postscript(file='plots_final/log2FC_KS1-KS2-days.eps', width=2.26, height=1.57)
cairo_pdf(filename='plots_final/log2FC_KS1-KS2-days.pdf', width=2.26, height=1.57, 
         onefile=FALSE, fallback_resolution = 600)

par(mar=c(2,2,0.3,0.3), mgp=c(0.5,0,0))
plot(shared.df$log2FC_KS2, shared.df$log2FC_KS1,
     pch=16, cex=0.4, col=alpha('gray60',0.3),
     xlim=c(-4,6), ylim=c(-4,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/MGI.chon.rda')

points(shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS2,
       shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS1,
       pch=16, cex=0.4, col=alpha('orange',0.7))

points.chon <- shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1 & shared.df$ensembl %in% MGI.chon$gene_id),]

points(points.chon$log2FC_KS2, points.chon$log2FC_KS1,
       pch=16, cex=0.4, col='deepskyblue')

axis(1, seq(-8,8, by=1), tck=-0.02, cex.axis=0.7)
axis(2, seq(-8,8, by=1), tck=-0.015, cex.axis=0.7)
mtext('log2FC, KS2 D14 vs D7', side=1, line=1, adj=0.5, cex=0.7)
mtext('log2FC, KS1 D14 vs D7', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
rect(2,3,6,8,col=NA,border='black')
dev.off()
```

Fig 5F inset:
```{r}
# setEPS()
# postscript(file='plots_final/log2FC_KS1-KS2-days-inset.eps', width=2.26, height=1.57)
cairo_pdf(filename='plots_final/log2FC_KS1-KS2-days-inset.pdf', width=2.26, height=1.57, 
         onefile=FALSE, fallback_resolution = 600)
par(mar=c(2,0.3,0.3,1), mgp=c(0.5,0,0))
plot(shared.df$log2FC_KS2, shared.df$log2FC_KS1,
     pch=16, cex=0.4, col=alpha('gray60',0.3),
     xlim=c(2,6), ylim=c(3,8),
     xlab = '', ylab='', axes=FALSE, 
     main='')

load(file='~/Desktop/temp/Fahrner lab/Data/KS2/220506_NGS/fahrner_kabuki2/RNAseq/final_analysis/MGI.chon.rda')

points(shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS2,
       shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1),]$log2FC_KS1,
       pch=16, cex=0.4, col=alpha('orange',0.7))

points.chon <- shared.df[which(shared.df$padj_KS2<0.1 & shared.df$padj_KS1<0.1 & shared.df$ensembl %in% MGI.chon$gene_id),]

points(points.chon$log2FC_KS2, points.chon$log2FC_KS1,
       pch=16, cex=0.4, col='deepskyblue')

int.genes <- c('Sulf1','Dcn','Fgf9','Gpnmb','Fgf1','Vnn1','Chad','Col10a1','Col13a1','Bglap','Bglap2')
# int.genes <- c('Sulf1','Dcn','Fgf9','Gpnmb','Fgf1','Chad','Col10a1','Bglap','Bglap2')

points(shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS2,
       shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS1,
       pch = 1, cex = 1, col = 'black')

xJitter <- c(-0.7,-0.2,0.2,0.2,0.2,-0.7,0.2,-0.8,0,0.2,-0.9)
yJitter <- c(0,0.5,0,0,0,0.2,0,-0.5,0.5,0,0)

text(shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS2+xJitter,
     shared.df[which(shared.df$mgi_symbol %in% int.genes),]$log2FC_KS1+yJitter,
     adj=0, cex = 0.6, col='black',font=3,
     labels=shared.df[which(shared.df$mgi_symbol %in% int.genes),]$mgi_symbol)

axis(1, seq(2,6, by=1), tck=-0.02, cex.axis=0.7)
axis(4, seq(3,8, by=1), tck=-0.015, cex.axis=0.7)
mtext('', side=1, line=1, adj=0.5, cex=0.7)
mtext('', side=2, line=1, adj=0.5, cex=0.7)
box(col='black')
abline(h=0, lty=2, col='red')
abline(v=0, lty=2, col='red')
abline(a=0, b=1, col='blue')
dev.off()
```

Scratch space
```{r}
sv.var <- data.frame(sv1=svseq$sv[,1], sv2=svseq$sv[,2], 
                     genotype=dds.coll.filtered$genotype, 
                     kabuki=dds.coll.filtered$kabuki)
sv.var$group <- interaction(sv.var$genotype, sv.var$kabuki)
sv.var$group <- factor(sv.var$group, levels=c('Mut.KS2','Wt.KS1','Wt.KS2'))
group.colors <- c('#FB0207','#0F80FF','#0F80FF')
group.shapes <- c(17,16,17)

ggplot(data=sv.var, aes(x=sv1, y=sv2, color=group, shape=group)) +
  geom_point(size=3) +
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())

batch <- c(rep(1,3), rep(2,3), rep(1,3))
group <- c(rep(1,3), rep(2,6))
adj.counts <- ComBat_seq(counts(dds.coll.filtered), batch=batch, group=group)
pca <- prcomp(adj.counts)
percentVar <- pca$sdev^2/sum(pca$sdev^2)*100
pca <- data.frame(pca$rotation, genotype=dds.coll.filtered$genotype, kabuki=dds.coll.filtered$kabuki)
pca$group <- interaction(pca$genotype, pca$kabuki)
pca$group <- factor(pca$group, levels=c('Mut.KS2','Wt.KS1','Wt.KS2'))
group.colors <- c('#FB0207','#0F80FF','#0F80FF')
group.shapes <- c(17,16,17)
ggplot(data=pca, aes(x=PC1, y=PC2, color=group, shape=group)) +
  geom_point(size=3) +
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) +
  xlab('PC1: 90.3% variance') +
  ylab('PC2: 4.1% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())
```

Testing out Combat-Seq
```{r}
col.data <- as.data.frame(colData(dds.coll.filtered))
col.data$batch <- ifelse(grepl("KS1",col.data$kabuki),"1","2")
col.data$group <- with(col.data, ifelse(
  genotype=="Mut" & kabuki=="KS1" & day=="D7", "1", ifelse(
    genotype=="Mut" & kabuki=="KS1" & day=="D14", "2", ifelse(
      genotype=="Mut" & kabuki=="KS2" & day=="D7", "3", ifelse(
        genotype=="Mut" & kabuki=="KS2" & day=="D14", "4", ifelse(
          genotype=="Wt" & day=="D7", "5", ifelse(
            genotype=="Wt" & day=="D14", "6","")))))))
adj.counts <- ComBat_seq(counts(dds.coll.filtered), batch=col.data$batch, group=col.data$group)
pca <- prcomp(adj.counts)
percentVar <- pca$sdev^2/sum(pca$sdev^2)*100
pca <- data.frame(pca$rotation, 
                     genotype=test$genotype, 
                     kabuki=test$kabuki,
                     day=test$day)
pca$group <- interaction(pca$genotype, pca$kabuki, pca$day)
pca$group <- factor(pca$group, levels=as.vector(unique(pca$group)))
group.colors <- c('#FB0207','#FB0207','#0F80FF','#0F80FF',
                  '#FB0207','#FB0207','#0F80FF','#0F80FF')
group.shapes <- c(17,16,2,1,2,1,17,16)
ggplot(data=pca, aes(x=PC1, y=PC2, color=group, shape=group)) +
  geom_point(size=3) +
  scale_shape_manual(values=group.shapes) +
  scale_color_manual(values=group.colors) +
  xlab('PC1: 70.0% variance') +
  ylab('PC2: 25.4% variance') +
  theme_bw(base_size=12) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())

adj.counts.df <- as.data.frame(adj.counts)
genes <- rownames(adj.counts.df)
rownames(adj.counts.df) <- NULL
adj.counts.df <- cbind(genes,adj.counts.df)
dds.adj <- DESeqDataSetFromMatrix(countData=adj.counts.df, 
                                  colData=colData(dds.coll.filtered), 
                                  design=~genotype*kabuki*day, 
                                  tidy=TRUE)
```